src/simplewallet/simplewallet.cpp

```diff
@@ -1462,6 +1464,72 @@ bool simple_wallet::handle_command_line(const boost::program_options::variables_
   return true;
 }
 //----------------------------------------------------------------------------------------------------
+bool simple_wallet::prompt_for_new_node()
+{
+    bool connect = false;
+    std::string node_url;
+
+    message_writer() << tr("Do you want to connect to a remote node?");
+    std::string confirm = command_line::input_line(tr("(Y/Yes/N/No): "));
+    if(std::cin.eof())
+    {
+      LOG_ERROR("Unexpected std::cin.eof() - Exited simple_wallet::prompt_for_new_node()");
+      return false;
+    }
+    connect = command_line::is_yes(confirm);
+
+    if (connect) {
+        message_writer() << tr("Please enter remote node: ");
+        node_url = command_line::input_line(tr("http(s)://[domain]:[port] (default port 18081): "));
+
+        boost::regex rgx("^(.*)://([A-Za-z0-9\\-\\.]+)(:[0-9]+)?(.*)");
+        boost::cmatch match;
+        if (boost::regex_match(node_url.c_str(), match, rgx)) {
+            // If no port has been entered, use the default
+            if (!match[3].length()) {
+                node_url = match[1] + std::string("://") + match[2] + std::string(":18081");
+            }
+        }
+
+        if (match.size() >= 2) {
+            if (!std::string("https").compare(match[1])) {
+                message_writer() << tr("Attention: Nodes not supporting 'https://' will transmit data unencrypted.\n"
+                                       "Do you still want to connect to this node?");
+
+                confirm = command_line::input_line(tr("(Y/Yes/N/No): "));
+                if(std::cin.eof())
+                {
+                  LOG_ERROR("Unexpected std::cin.eof() - Exited simple_wallet::prompt_for_new_node()");
+                  return false;
+                }
+                connect = command_line::is_yes(confirm);
+            }
+        }
+    } else {
+        message_writer() << tr("Do you want to connect to your local node?");
+        confirm = command_line::input_line(tr("(Y/Yes/N/No): "));
+        if(std::cin.eof())
+        {
+          LOG_ERROR("Unexpected std::cin.eof() - Exited simple_wallet::prompt_for_new_node()");
+          return false;
+        }
+        connect = command_line::is_yes(confirm);
+
+        if (connect) {
+            int daemon_port = m_wallet->testnet() ? config::testnet::RPC_DEFAULT_PORT : config::RPC_DEFAULT_PORT;
+            node_url = std::string("http://localhost:") + std::to_string(daemon_port);
+        }
+    }
```

Comment:
This code split seems entirely unnecessary, no ? The first one doesn't allow for testnet too.

Reply:
Would it not depend on the node being testnet? That is what I assumed, otherwise I will adapt the code of course.

Reply:
I don't understand your question. My comment meant that one of the code branches handles testnet, the other does not.

Reply:
Should be fine? The `--testnet` flag already handles this on startup, I'd assume. I just check if it is testnet or not if a local node is selected as I can then dynamically set the default port from the config.